src/main/java/kg/jarkyn/server/RequestingSocket.java

```diff
@@ -0,0 +1,33 @@
+package kg.jarkyn.server;
+
+import kg.jarkyn.server.Incoming.Requester;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+
+public class RequestingSocket implements Requester {
```

Comment:
well, I have to pick a point of view, right now I am choosing to look at everything from a point of view of server + request, as far as it is concerned client socket is the requesting one
plus requesting does not exclude being able to receive what we sent to it. I will roll with this for now, but this is not final